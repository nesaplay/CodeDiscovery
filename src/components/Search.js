import React from 'react'
import { Input, Button } from 'semantic-ui-react'

const Search = ({ 
    searchHandler, 
    clickHandler,
    isLoading 
}) => (
        <Input icon='github'
            iconPosition='left' 
            placeholder='Search github repositiories...' 
            size='huge'
            fluid
            onChange={({ target }) => searchHandler(target.value)}
            action={
                <Button type='submit' 
                    loading={isLoading} 
                    onClick={() => clickHandler()}>
                    Search
                </Button>
            }
        />
    )

export default Search