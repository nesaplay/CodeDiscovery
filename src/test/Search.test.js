import React from 'react'
import Search from '../components/Search'

it('renders a component', () => {
	const wrapper = shallow(
        <Search />
    )

	expect(wrapper).toMatchSnapshot()
})
