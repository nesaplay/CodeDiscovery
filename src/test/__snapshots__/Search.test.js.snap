// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a component 1`] = `
<Input
  action={
    <Button
      as="button"
      loading={undefined}
      onClick={[Function]}
      type="submit"
    >
      Search
    </Button>
  }
  fluid={true}
  icon="github"
  iconPosition="left"
  onChange={[Function]}
  placeholder="Search github repositiories..."
  size="huge"
  type="text"
/>
`;
